src/circuits/gates.jl

```diff
@@ -302,97 +302,16 @@ gate(::GateName"CCCNOT") =
 # Random Haard unitary:
 # 
 # Reference: http://math.mit.edu/~edelman/publications/random_matrix_theory.pdf
-function gate(::GateName"randU", N::Int = 2;
+function gate(::GateName"randU", N::Int = 1;
```

Comment:
I remembered why I didn't design it with `N` as the number of qubits/sites: these gate definitions may be useful beyond just qubit degrees of freedom, such as qudits (specific examples would be Electron-type sites where it is natural to group the up and down spins together). I know we don't necessarily have qudits in mind right now, but it is nice to design interfaces to be "future-proof", i.e. make them general enough to accommodate reasonable generalizations. But I guess in that case we could add another keyword argument `dim = 2` where you could specify the dimension of the local Hilbert space, and that could be determine in the outer `gate` definition from the dimensions of the indices.

Reply:
Sounds good, so no need to chance anything on this now right? We can add that kwarg when we need, if so.